### ISSUE TITLE
Normalize all user-provided functions to promise via utility method internally

### ISSUE THREAD

**Issue by scotthovestadt** [dthree/vorpal#102]
To ensure that all async user-provided functions can use either a callback or a promise, a utility method should be created that normalizes everything into returning a promise. It'll simplify the code.

We may also want to consider using Babel's async/await to further simplify the code.

@dthree Thoughts?

**Comment by dthree** [comment]
I agree with standardizing it. Additionally, it would be good to consistently permit an optional sync response as well (simply by the user not passing in a callback to the method).

However, isn't async/await ES7 experimental? Not sure I want to get into that.

**Comment by scotthovestadt** [comment]
Yeah, it's ES7... but probably the most widely used experimental feature in the history of JavaScript. You should play with it. It changes everything.

You can setup Babel to compile it to generators, which are a part of NodeJS now.

**Comment by dthree** [comment]
I'll check it out.
